sdk/search/azure-search/src/main/java/com/azure/search/models/SearchResult.java

```diff
@@ -45,8 +44,7 @@
      *
      * @return the additionalProperties value.
      */
-    @JsonIgnore
-    public Document getDocument() {
+    public Map<String, Object> getDocument() {
```

Comment:
This is a public contract -- Shouldn't the return type be `SearchDocument` (renamed from `Document`)?

Reply:
Yes. This is manually craft code. So Alan's PR will address this.

Reply:
Made the changes.